Updated vendor alertmanager panics

Currently the alertmanager panics when run. 
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xc2b640]

goroutine 435 [running]:
github.com/cortexproject/cortex/vendor/github.com/prometheus/alertmanager/types.(*memMarker).registerMetrics(0xc000c6f140, 0x0, 0x0)
	/go/src/github.com/cortexproject/cortex/vendor/github.com/prometheus/alertmanager/types/types.go:125 +0x120
github.com/cortexproject/cortex/vendor/github.com/prometheus/alertmanager/types.NewMarker(...)
	/go/src/github.com/cortexproject/cortex/vendor/github.com/prometheus/alertmanager/types/types.go:97
github.com/cortexproject/cortex/pkg/alertmanager.New(0xc000c135e0, 0xc000c135e0, 0x34e0ce0, 0xc000896840)
	/go/src/github.com/cortexproject/cortex/pkg/alertmanager/alertmanager.go:101 +0x6d0
github.com/cortexproject/cortex/pkg/alertmanager.(*MultitenantAlertmanager).newAlertmanager(0xc000982120, 0xc000c79038, 0x4, 0xc0004ca230, 0x0, 0x0, 0xc0009856e0)
	/go/src/github.com/cortexproject/cortex/pkg/alertmanager/multitenant.go:445 +0xfe
github.com/cortexproject/cortex/pkg/alertmanager.(*MultitenantAlertmanager).setConfig(0xc000982120, 0xc000a16740, 0x4, 0x4)
	/go/src/github.com/cortexproject/cortex/pkg/alertmanager/multitenant.go:406 +0xcca
github.com/cortexproject/cortex/pkg/alertmanager.(*MultitenantAlertmanager).addNewConfigs(0xc000982120, 0xc00020a308, 0x1, 0x1)
	/go/src/github.com/cortexproject/cortex/pkg/alertmanager/multitenant.go:305 +0x1fa
github.com/cortexproject/cortex/pkg/alertmanager.(*MultitenantAlertmanager).Run(0xc000982120)
	/go/src/github.com/cortexproject/cortex/pkg/alertmanager/multitenant.go:233 +0x82
created by github.com/cortexproject/cortex/pkg/cortex.(*Cortex).initAlertmanager
	/go/src/github.com/cortexproject/cortex/pkg/cortex/modules.go:383 +0xea
```

This is caused by the following call:
```go
	am.marker = types.NewMarker(nil)
```

New marker can not take a nil registry or else it will panic.

--- a/pkg/alertmanager/alertmanager.go
+++ b/pkg/alertmanager/alertmanager.go
@@ -98,7 +98,7 @@ func New(cfg *Config) (*Alertmanager, error) {
 		am.nflog.SetBroadcast(c.Broadcast)
 	}
 
-	am.marker = types.NewMarker(nil)
+	am.marker = types.NewMarker(am.registry)
 
 	silencesID := fmt.Sprintf("silences:%s", cfg.UserID)
 	am.silences, err = silence.New(silence.Options{
@@ -121,8 +121,7 @@ func New(cfg *Config) (*Alertmanager, error) {
 		am.wg.Done()
 	}()
 
-	marker := types.NewMarker(nil)
-	am.alerts, err = mem.NewAlerts(context.Background(), marker, 30*time.Minute, am.logger)
+	am.alerts, err = mem.NewAlerts(context.Background(), am.marker, 30*time.Minute, am.logger)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create alerts: %v", err)
 	}
@@ -130,7 +129,7 @@ func New(cfg *Config) (*Alertmanager, error) {
 	am.api, err = api.New(api.Options{
 		Alerts:     am.alerts,
 		Silences:   am.silences,
-		StatusFunc: marker.Status,
+		StatusFunc: am.marker.Status,
 		Peer:       cfg.Peer,
 		Logger:     log.With(am.logger, "component", "api"),
 		GroupFunc: func(f1 func(*dispatch.Route) bool, f2 func(*types.Alert, time.Time) bool) (dispatch.AlertGroups, map[model.Fingerprint][]string) {

